### PR TITLE
Add 'PairIJ Coeffs' section to the list of sections in LAMMPS parser

### PR DESCRIPTION
**Pull request by mglagolev**
When called with "pair ij" option, the write_data command in LAMMPS writes a data file with a 'PairIJ Coeffs' section, as stated https://docs.lammps.org/write_data.html
Absence of this section in the SECTIONS list of LAMMPSParser.py can lead to an error.
This fix ensures correct processing of LAMMPS data files with a 'PairIJ Coeffs' section.

Changes made in this Pull Request:

'PairIJ Coeffs' added to the list of sections in LAMMPSParser.py

### CODE REVIEW

**Comment by mglagolev** [comment]
This should actually solve the issue #3336

**Comment by hmacdope** [comment]
Hi @mglagolev thanks for the contribution! 

Would you be able to add a test with a small example test file to show the issue is resolved.

You will also need to update `CHANGELOG`, and add yourself to `AUTHORS`.

 I would also ask you to possibly introduce yourself on the [developer mailing list](https://groups.google.com/g/mdnalysis-devel?pli=1) to welcome you to the community and have a means of contacting you. :)

**Comment by mglagolev** [comment]
@hmacdope Thank you for the suggestions! I pushed the changes to AUTHORS and CHANGELOG to my branch, shall I do anything else? This is my first pull request ever.
Shall I add the test file and test script to my branch also? If so, where shall I put it?

**Comment by mglagolev** [comment]
Added a datafile and a reading script to testsuite/MDAnalysisTests/topology

File: testsuite/MDAnalysisTests/topology/PR3959_test.py
Comment (hmacdope) [inline testsuite/MDAnalysisTests/topology/PR3959_test.py]: This can become part of the other LAMMPS Topology tests in `test_lammpsdata.py` it doesn't need its own file.

File: testsuite/MDAnalysisTests/topology/PR3959_test.py
Comment (hmacdope) [inline testsuite/MDAnalysisTests/topology/PR3959_test.py]: This needs to be added as a `datafile` see datafiles.py on how to do that.

File: testsuite/MDAnalysisTests/topology/PR3959_test.data
Comment (hmacdope) [inline testsuite/MDAnalysisTests/topology/PR3959_test.data]: This needs to be moved to `data` and added to datafiles.py.

**Review by hmacdope — CHANGES_REQUESTED** [review]
A good start :) See my comments. Ping me if unsure of anything.

**Comment by mglagolev** [comment]
@hmacdope Thank you for your guiding! I tried to replicate what I saw in test_lammpsdata.py, and committed to the branch. When I run the test:

cd testsuite/MDAnalysisTests
pytest  --disable-pytest-warnings

It shows errors in some other modules, but not topology.
However, I'm not sure that I've done everything correctly and the testing actually takes place.

I'm also wondering, shall my datafile state the number of impropers and charges, because these are mentioned as expected_attrs in LammpsBase.

**Comment by hmacdope** [comment]
> @hmacdope Thank you for your guiding! I tried to replicate what I saw in test_lammpsdata.py, and committed to the branch. When I run the test:

well done on getting the test files sorted out! 
> cd testsuite/MDAnalysisTests pytest --disable-pytest-warnings
> 
> It shows errors in some other modules, but not topology. However, I'm not sure that I've done everything correctly and the testing actually takes place.

1. The errors are coming from the issue with the density code, #3976 will fix it in due time so no need to worry about it yet.

~~You have imported the file but the testing is not yet taking place as you need to have `asserts`, but you have a good start~~



~~I will review to show you how to make the testing work~~

EDIT: Looks like the testing is working, sorry I didn't see that it was being done in the base class.

> I'm also wondering, shall my datafile state the number of impropers and charges, because these are mentioned as expected_attrs in LammpsBase.

~~Lets see what happens when we add testing.~~

**Comment by hmacdope** [comment]
To test only your module go to `MDAnalysistests/topology/` and run `pytests test_lammpsdata.py`

File: package/CHANGELOG
Comment (hmacdope) [inline package/CHANGELOG:21]: You will need to merge with upstream and move this to 2.5.0

Should be 

```
git fetch upstream
git merge upstream/develo
# fix any merge conflicts if they arise.
```

**Review by hmacdope — CHANGES_REQUESTED** [review]

**Comment by mglagolev** [comment]
> To test only your module go to `MDAnalysistests/topology/` and run `pytests test_lammpsdata.py`

Done that.
My file understandably fails on test_expected_attributes:  AssertionError: Missing expected attribute: charges
What shall I do? Create a different file with a different LAMMPS atom_style, which will include charges?

File: package/CHANGELOG
Comment (mglagolev) [reply package/CHANGELOG:21]: Thanks, merge also done. Shall I do a push after the merge?

File: testsuite/MDAnalysisTests/data/lammps/pairij_coeffs.data
Comment (IAlibay) [inline testsuite/MDAnalysisTests/data/lammps/pairij_coeffs.data]: Would it be possible to gzip or bzip2 this? i.e. is this something we support (I know we do for PDBs and similar, so I assume it should work here too?)

**Review by IAlibay — CHANGES_REQUESTED** [review]
I'm going to make that last one blocking, I'll unblock if it turns out to not be feasible.

File: testsuite/MDAnalysisTests/data/lammps/pairij_coeffs.data
Comment (mglagolev) [reply testsuite/MDAnalysisTests/data/lammps/pairij_coeffs.data]: @IAlibay I'm sorry, I'm lost a little bit. How can I help at this point?

Comment (orbeckst) [reply testsuite/MDAnalysisTests/data/lammps/pairij_coeffs.data]: Compress the file(s) with `bzip2` and store it as `testsuite/MDAnalysisTests/data/lammps/pairij_coeffs.data.bz2`. MDAnalysis will just read it the compressed file and we use up less space in the package and on disk.

File: package/CHANGELOG
Comment (orbeckst) [reply package/CHANGELOG:21]: Yes, just keep pushing to your branch and this PR.

File: testsuite/MDAnalysisTests/data/lammps/pairij_coeffs.data
Comment (mglagolev) [reply testsuite/MDAnalysisTests/data/lammps/pairij_coeffs.data]: Thank you! Recommitted with a zipped file.

**Comment by mglagolev** [comment]
Thank you all, I've reverted the previous commit and recommitted with a zipped file.

File: package/CHANGELOG
Comment (IAlibay) [reply package/CHANGELOG:21]: As per @hmacdope's comment - could you move this up to the 2.5.0 entry please ? You'll also need to add yourself to the user list there.

File: testsuite/MDAnalysisTests/topology/test_lammpsdata.py
Comment (IAlibay) [inline testsuite/MDAnalysisTests/topology/test_lammpsdata.py:181]:
```suggestion


```

File: testsuite/MDAnalysisTests/topology/test_lammpsdata.py
Comment (IAlibay) [inline testsuite/MDAnalysisTests/topology/test_lammpsdata.py:201]:
```suggestion


```
As above - this was picked up by flake8 as a PEP8 violation

File: testsuite/MDAnalysisTests/topology/test_lammpsdata.py
Comment (IAlibay) [inline testsuite/MDAnalysisTests/topology/test_lammpsdata.py:183]: Since you're subclassing `LammpsBase` but your expected attributes don't include everything (in this case charges), could you should just redefine `expected_attrs` in the subclass to match the attributes contained in this new file?

**Review by IAlibay — CHANGES_REQUESTED** [review]
Couple things from my side of things, thanks for bzipping that datafile.

File: testsuite/MDAnalysisTests/topology/test_lammpsdata.py
Comment (mglagolev) [reply testsuite/MDAnalysisTests/topology/test_lammpsdata.py:183]: @IAlibay Thank you for pointing everything out. I've addressed the issues.

**Comment by hmacdope** [comment]
@mglagolev If you could fix the failing tests and the merge conflicts, we can move forward from there. 
 :)

File: package/CHANGELOG
Comment (IAlibay) [inline package/CHANGELOG]:
```suggestion
```
Please don't add the extra line here (we try to keep the format of this file reasonably strict to make life easier on releases).

**Review by IAlibay — CHANGES_REQUESTED** [review]
Just the one quick thing from me.

**Review by IAlibay — APPROVED** [review]
Thanks!

**Comment by mglagolev** [comment]
@IAlibay @hmacdope 
It seems that everything should be fine now, except for the darker lint test. But I thought maybe I should comply with the style already used in test_lammpsdata.py.
Thank you all for your big help with this small commit!

**Comment by orbeckst** [comment]
Thank you for your contribution @mglagolev !

> Thank you all for your big help with this small commit!

Yes, as you can see, when working on software that's used by a few thousand researchers, there's a lot more to do for maintaining quality than just to add a few lines of code and hoping it will be fine. We do all this reviewing and insisting on tests with the goal to keep MDAnalysis a useful and maintainable piece of software for the foreseeable future.

File: testsuite/MDAnalysisTests/topology/test_lammpsdata.py
Comment (IAlibay) [inline testsuite/MDAnalysisTests/topology/test_lammpsdata.py]:
```suggestion

```
No worries about the darker linting, as long as it's not breaching PEP8 it ends up just `black` trying to enforce its style.

That being said the above "whitespace on an empty line" is a PEP8 violation (I suspect the suggested edit here won't work well, essentially the empty line is fine, but it can't have any whitespaces on it)

Sorry about this, as @orbeckst explains re: keeping maintenance costs down, we end up requiring a lot of weird formatting rules that do tend to be rather unfriendly to new contributors :(

**Comment by mglagolev** [comment]
@orbeckst @IAlibay @hmacdope 
I absolutely understand that this rigorousness is essential, and I'm very grateful for your immense work on this project.
Thanks for sort of onboarding me. I hope that I can contribute more in the future.

**Review by hmacdope — APPROVED** [review]
Fantastic work @mglagolev! Thanks for the contribution :smile:  Many LAMMPS users will appreciate this one I'm sure. :+1:

**Pull request merged**
